src/core/lib/layer-manager.js

```diff
@@ -214,19 +230,23 @@ export default class LayerManager {
     }
 
     if ('layerFilter' in parameters) {
-      this.context.layerFilter = parameters.layerFilter;
-      if (this.context.layerFilter !== parameters.layerFilter) {
+      if (this.layerFilter !== parameters.layerFilter) {
+        this.layerFilter = parameters.layerFilter;
         this.setNeedsRedraw('layerFilter changed');
       }
     }
 
     if ('drawPickingColors' in parameters) {
-      if (this.context.drawPickingColors !== parameters.drawPickingColors) {
+      if (parameters.drawPickingColors !== this.drawPickingColors) {
+        this.drawPickingColors = parameters.drawPickingColors;
         this.setNeedsRedraw('drawPickingColors changed');
       }
     }
 
-    Object.assign(this.context, parameters);
+    // A way for apps to add data to context to access in layers
+    if ('context' in parameters) {
+      Object.assign(this.context.userData, parameters.context);
```

Comment:
Does this mean that old `userData` fields may never go away?

Reply:
Good point. Updated the code to just link to the provided data object. This is a new field so no risk of breaking.